libenv/unix_iface.c

```diff
@@ -704,8 +704,13 @@ static void FindV6InterfacesInfo(EvalContext *ctx, Rlist **interfaces, Rlist **h
                     xsnprintf(prefixed_ip, sizeof(prefixed_ip), IPV6_PREFIX"%s", ip->name);
                     EvalContextClassPutHard(ctx, prefixed_ip, "inventory,attribute_name=none,source=agent");
 
-                    // Add IPv6 address to sys.ip_addresses
-                    RlistAppendScalar(ips, ip->name);
+                    if (current_interface[0] != '\0'
+                        && !IgnoreInterface(current_interface))
```

Comment:
If I have understood this correctly, it needs to be its own if test, and not bundled in the if below, due to the extra condition present there.

Reply:
Yes, but all of this should be inside the same if (All IPv6 info should respect ignored interfaces).

And, you can just make a special if to avoid adding the interface twice, don't repeat the shared condition.